qiskit/utils/backend_utils.py

```diff
@@ -137,14 +137,16 @@ def is_statevector_backend(backend):
     Returns:
         bool: True is statevector
     """
+    if backend is None:
+        return False
     if has_aer():
         from qiskit.providers.aer.backends import AerSimulator, StatevectorSimulator
 
         if isinstance(backend, StatevectorSimulator):
             return True
-        if isinstance(backend, AerSimulator) and backend.name() == "aer_simulator_statevector":
+        if isinstance(backend, AerSimulator) and has_save_statevector(backend):
```

Comment:
This does not return True if is a statevector simulator - it returns True if it has the _potential_ to be used a statevector simulator. There is a big difference - its the reason we do not do it that way.

The original issue had a link to a broader issue around improving QuantumInstance and the way the algos work with backends. We had some out-of-band discussion with Chris over Slack around the bigger picture - the solution that was merged was the best we could do at the time.